### PR TITLE
JIT: Fix some cases using `BasicBlock::bbFallsThrough`

### PR DESCRIPTION
**Pull request by jakobbotsch**
`bbFallsThrough` still returns true for `BBJ_COND`; we have a couple of places using it as a "control flows from prev block" check, which is wrong after #97488.
Should fix the test failures I'm seeing over in #97182.

### CODE REVIEW

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -13886,6 +13886,10 @@ void Compiler::fgMorphBlock(BasicBlock* block)
                             assertionsOut = pred->bbAssertionOutIfFalse;
                         }
                     }
+                    else
+                    {
+                        assertionsOut = pred->bbAssertionOut;
+                    }
```
Comment (jakobbotsch) [inline src/coreclr/jit/morph.cpp:13892]: Small drive-by fix for some UB here (in the `useCondAssertions` case we were accessing an inactive union member).

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib PTAL @amanasifkhalid 

[Diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=546039&view=ms.vss-build-web.run-extensions-tab). A few cases where we now recognize an induction due to `optExtractInitTestIncr` being more general, and then start loop cloning.
I left `bbFallsThrough` alone since I wasn't sure if we want to change its behavior yet (noticed a few cases that explicitly check it and then filter out `BBJ_COND` cases).

**Review by amanasifkhalid — APPROVED** [review]

**Comment by amanasifkhalid** [comment]
Thanks for the fix!

> I left bbFallsThrough alone since I wasn't sure if we want to change its behavior yet (noticed a few cases that explicitly check it and then filter out BBJ_COND cases).

My hope is that we can get rid of `bbFallsThrough` altogether soon, and replace it with a pattern like the unique predecessor check you're using here. Those explicit `BBJ_COND` checks were probably introduced as quirks to reduce diffs for now.
